### ISSUE TITLE
How about gpio in/out command via AT ? thank's

### ISSUE THREAD

**Issue by gorec2005** [igrr/atproto#13]
subj...

**Comment by phil31** [comment]
please clarify your question !

**Comment by igrr** [comment]
@gorec2005 You mean like `AT+GPIO1=1` to set GPIO1 high and `AT+GPIO2=?` to read GPIO2?

**Comment by gorec2005** [comment]
whether it is possible to add the ability through AT commands control the state of GPIO outputs and receive status GPIO inputs

**Comment by gorec2005** [comment]
Yes!

**Comment by igrr** [comment]
Yep it is possible.
I wonder though why would you want this, are you connecting the esp8266 to some microcontroller without enough GPIOs?

**Comment by gorec2005** [comment]
if use esp without microcontroller! - may be just for powerplug ( tcp-server+gpio-in-out )...
if it can do onewire protocol it can more useful things without microcontroller :-)

**Comment by igrr** [comment]
Ooh, I see.
Currently AT protocol is used on the serial (UART) interface, not on the
network (TCP) side. Data from network is passed through to the application
listening on the UART port, no parsing or command processing takes place
there.

You can definitely use AT protocol for network communications. You can even
reuse all my DCE code for that. You need to set up a tcp server in
user_init and then pump received data into dce_handle_input function as I
do with serial.

**Comment by gorec2005** [comment]
Yes! you right!!! with this posibility it's will  be really cool for home automation!!!
